### ISSUE TITLE
Add proxy class that can implement interfaces at runtime.

### ISSUE THREAD

**Issue by justinfagnani** [dart-lang/sdk#5641]
In unchecked mode it's possible to create objects that implement an interface via noSuchMethod and assign them to a variable or argument of the interface type. This is useful for things like typed proxies for JS interop, automatic stubs for remote/isolated services, etc. In checked mode though assigning to a typed variable causes an exception.

For Proxy, this should work:

var proxy = new Proxy([Foo, Bar]);
assert(proxy is Bar);
assert(proxy is Foo);

Then subclasses can implement behavior.

**Comment by pulyaevskiy** [comment]
Hey team, just curious, the fact that it's not `accepted` anymore means this will not be implemented?

I am really interested in solution for this use case.

**Comment by floitschG** [comment]
@pulyaevskiy:
Sorry for the late response.
As long as the bug is open, it hasn't been rejected. It's not yet high on our TODO lists, and, even when accepted, the outcome might look very different from the proposal here.

TL;DR: we see the need, but we are not currently working on it.
